server/src/main/java/org/opensearch/rest/action/cat/AbstractCatAction.java

```diff
@@ -98,4 +99,12 @@ protected Set<String> responseParams() {
         return RESPONSE_PARAMS;
     }
 
+    /**
+     * Method to check if limits defined in {@link RequestLimitSettings} are applicable to an action.
+     *
+     * @return True / False status
+     */
+    protected boolean isRequestLimitCheckSupported() {
```

Comment:
Pagination API will extend the existing cat action so we wanted to enforce it at action level. But, we wanted to provide a default support in Abstract class so that others can override the actions.